packages/engine/Source/DataSources/SampledProperty.js

```diff
@@ -674,6 +674,30 @@ SampledProperty.prototype.addSamples = function (
   this._definitionChanged.raiseEvent(this);
 };
 
+/**
+ * Gets the time of the sample. Negative for backward.
+ *
+ * @param {number} index The index of samples list.
+ * @returns {JulianDate | undefined} The JulianDate time of the sample, or undefined if failed.
+ */
+SampledProperty.prototype.getSample = function (index) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.defined("index", index);
```

Comment:
@lukemckinstry Hi, I think it is better to allow an out of range index and return undefined. Because it is the default behavior of [Array.prototype.at](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at). And attempting to access an index outside the array range will also return undefined. 

```js
;[0, 1, 2][3] // => undefined
```

In CesiumJS, types that use the stack data structure, such as `EntityCollection.values`, also use `Array` directly, so the default behavior that conforms to arrays may be more intuitive.

Reply:
makes sense to me :+1: